### DISCUSSION TITLE
Jinja Templating in an operator

### DISCUSSION THREAD

**Discussion by tagoad** [apache/airflow#24081] [Q&A] [answered]
Quick question that hopefully someone can help with. I am familiar basic jinja templating with operators and such, but I was wondering if someone had a good guide on using it within operators? I want to leverage jinja for some data manipulation but I need to run some validation on the data before executing the templating. Is there a way to do this? I would need to pass in variables and helper functions to be used in the template as well. I looked at the Jinja docs but stopped when it talked about environments because I wanted to see if I could leverage the existing jinja environment in airflow. Any suggestions/guidance would be appreciated!

**Comment by potiuk** [comment, marked as answer]
You can use context which is passed as "execute" parameter. But then the Jinja processing if it is to be done inside the "execute" should be done by you - and there is nothing wrong with that, just the usuall processing that you'd do with Jinja in Python. But the context is what could bind your internal Jinja processing with the one done by airflow before the templates are processed.

Just make sure the args/default args of your operator are not part of "templated_fields" as they woudl be processed by Airflow before your execute method is reached.

**Reply by tagoad** [reply]
Thanks for the reply! I was probably just reading to far into the Jinja docs then, it mentions not making multiple environments so i took that as i should try to leverage the existing one in airflow since its already there.
